### PR TITLE
Make room avatar and name colors configurable based on theme

### PR DESCRIPTION
**Pull request by chagai95**
## Type of change

- [X] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Make room avatar and name colors configurable based on theme. Only added dark and light for now.

## Motivation and context
Using a black room avatar results in black on black - but in general we would like to be able to customize based on customer needs.

## Tests

- Changed theme.
- Closed the app
- Reopened the app

## Tested devices

- [X] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [X] Changes has been tested on an Android device or Android emulator with API 21
- [X] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [X] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [X] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [X] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

Signed-off-by: Chagai Friedlander @me:chagai.website

### CODE REVIEW

**Comment by chagai95** [comment]
Just checking, did I do something wrong?

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/MatrixItemColorProvider.kt [hunk 1]
```diff
@@ -107,12 +105,28 @@ class MatrixItemColorProvider @Inject constructor(
             }
         }
 
-        @ColorRes
+        
+        @ColorInt
+        private fun getUserColorByIndexColorInt(colorProvider: ColorProvider, index: Int): Int {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/MatrixItemColorProvider.kt:110]: I think the fun above `getUserColorByIndex` could be removed.
And so the resources `R.color.element_name_0x` and `R.color.element_room_0x`.

**Review by bmarty — CHANGES_REQUESTED** [review]
Thanks for the PR.
The code does not compile, nor the test. Can you check please?

Also please note that the default room and user color are shared across all Element Matrix client. Since there is no change on the value here, this is fine to accept this change.

**Comment by bmarty** [comment]
Close as stale PR.

**Comment by chagai95** [comment]
> Close as stale PR.

Sorry, I don't work with matrix anymore in my daily join and didn't find the time in my free time to get around to this.

**Pull request closed without merging**